site/oauth2.md

```diff
@@ -0,0 +1,464 @@
+
+#### Audience must have/match the resource_server_id
+
+The `aud` ([Audience](https://tools.ietf.org/html/rfc7519#page-9)) identifies the recipients and/or resource_server of the JWT. By default, **RabbitMQ uses this field to validate the token** although we can disable it by setting `verify_aud` to `false`.  When it set to `true`, this attribute must either match the `resource_server_id` setting or in case of a list, it must contain the `resource_server_id`.
+
+
+### <a id="scope-translation" class="anchor" href="#scope-translation">Scope-to-Permission Translation</a>
+
+Scopes are translated into permission grants to RabbitMQ resources for the provided token.
+
+The current scope format is `&lt;permission>:&lt;vhost_pattern>/&lt;name_pattern>[/&lt;routing_key_pattern>]` where
+
+ * `<permission>` is an access permission (`configure`, `read`, or `write`)
+ * `<vhost_pattern>` is a wildcard pattern for vhosts token has access to.
+ * `<name_pattern>` is a wildcard pattern for resource name
+ * `&lt;routing_key_pattern>` is an optional wildcard pattern for routing key in topic authorization
+
+Wildcard patterns are strings with optional wildcard symbols `*` that match
+any sequence of characters.
+
+Wildcard patterns match as following:
+
+ * `*` matches any string
+ * `foo*` matches any string starting with a `foo`
+ * `*foo` matches any string ending with a `foo`
+ * `foo*bar` matches any string starting with a `foo` and ending with a `bar`
+
+There can be multiple wildcards in a pattern:
+
+ * `start*middle*end`
+ * `*before*after*`
+
+**To use special characters like `*`, `%`, or `/` in a wildcard pattern,
+the pattern must be [URL-encoded](https://en.wikipedia.org/wiki/Percent-encoding).**
+
+These are the typical permissions examples:
+
+- `read:*/*`(`read:*/*/*`) - read permissions to any resource on any vhost
+- `write:*/*`(`write:*/*/*`) - write permissions to any resource on any vhost
+- `read:vhost1/*`(`read:vhost1/*/*`) - read permissions to any resource on the `vhost1` vhost
+- `read:vhost1/some*` - read permissions to all the resources, starting with `some` on the `vhost1` vhost
+- `write:vhsot1/some*/routing*` - topic write permissions to publish to an exchange starting with `some` with a routing key starting with `routing`
+
+See the [wildcard matching test suite](./test/wildcard_match_SUITE.erl) and [scopes test suite](./test/scope_SUITE.erl) for more examples.
+
+Scopes should be prefixed with `resource_server_id`. For example,
+if `resource_server_id` is "my_rabbit", a scope to enable read from any vhost will
+be `my_rabbit.read:*/*`.
+
+### <a id="use-different-token-field" class="anchor" href="#use-different-token-field">Using a different token field for the Scope</a>
+
+By default the plugin will look for the `scope` key in the token, you can configure the plugin to also look in other fields using the `extra_scopes_source` setting. Values format accepted are scope as **string** or **list**
```

Comment:
By default the plugin will look for the `scope` key in the token. The plugin can be configured to also look in other fields using the `extra_scopes_source` setting. Accepted scope value formats are **string** or **list**